src/simplify_algebra.cpp

```diff
@@ -72,14 +74,15 @@ struct find_mul_conv
         auto a_ins    = r.instructions["a"];
         auto w_ins    = r.instructions["w"];
 
-        auto broadcast_op = any_cast<op::broadcast>(a_ins->get_operator());
-        if(broadcast_op.axis != 1)
+        auto a_strides = a_ins->get_shape().strides();
+        if(a_strides.at(0) != 0 or a_strides.at(1) != 1 or
+           std::any_of(
+               a_strides.cbegin() + 2, a_strides.cend(), [](std::size_t i) { return i != 0; }))
             return;
 
+        auto sq    = m.insert_instruction(ins, make_op("squeeze"), a_ins->inputs().front());
         auto new_a = m.insert_instruction(
-            ins,
-            make_op("broadcast", {{"axis", 0}, {"out_lens", w_ins->get_shape().lens()}}),
-            a_ins->inputs().front());
+            ins, make_op("broadcast", {{"axis", 0}, {"out_lens", w_ins->get_shape().lens()}}), sq);
```

Comment:
Is it possible to use `multibroadcast` here ? `broadcast` is deprecated.

Reply:
Possible but would need to squeeze and then unsqueeze to something like a [C, 1, 1] shape to use multibroadcast. Paul mentioned that it might be better to combine `broadcast` and `multibroadcast` into one operation.